app/views/hostgroups/index.html.erb

```diff
@@ -16,6 +18,14 @@
           <%= label_with_link(hostgroup, 150, authorizer) %>
         </td>
         <td>
+          <%= link_to hostgroup.hosts_count, hosts_path(:search => "hostgroup_fullname = #{hostgroup}") %>
+        </td>
+        <td>
+          <% if hostgroup.children.any? %>
```

Comment:
Hostgroups with no children have no data, so they can be left empty IMO.
I get no N+1 queries on my page